apps/E2E/src/Avatar/specs/Avatar.spec.win.ts

```diff
@@ -34,30 +29,47 @@ describe('Avatar Accessibility Testing', () => {
     await AvatarPageObject.scrollToTestElement();
   });
 
-  it('Validate accessibilityLabel', async () => {
-    await expect(await AvatarPageObject.getPrimaryComponentAttribute(ACCESSIBILITY_LABEL_ATTR)).toEqual(AVATAR_ACCESSIBILITY_LABEL);
+  it('Set accessibilityLabel -> Validate set accessibilityLabel is correct', async () => {
+    await expect(
+      await AvatarPageObject.compareAttribute(AvatarPageObject._primaryComponent, Attribute.AccessibilityLabel, AVATAR_ACCESSIBILITY_LABEL),
+    ).toBeTrue();
+
     await expect(await AvatarPageObject.didAssertPopup()).toBeFalsy(AvatarPageObject.ERRORMESSAGE_ASSERT);
   });
 
-  it('Validate accessibilityLabel from `name` prop', async () => {
-    await expect(await AvatarPageObject.getSecondaryComponentAttribute(ACCESSIBILITY_LABEL_ATTR)).toEqual(
-      AVATAR_ACCESSIBILITY_LABEL_BY_NAME,
-    );
+  it("Set 'name' prop -> Validate accessibilityLabel is correct", async () => {
+    await expect(
+      await AvatarPageObject.compareAttribute(
+        AvatarPageObject._secondaryComponent,
+        Attribute.AccessibilityLabel,
+        AVATAR_ACCESSIBILITY_LABEL_BY_NAME,
+      ),
+    ).toBeTrue();
+
     await expect(await AvatarPageObject.didAssertPopup()).toBeFalsy(AvatarPageObject.ERRORMESSAGE_ASSERT);
   });
 
-  it('Validate accessibilityHint', async () => {
-    await expect(await AvatarPageObject.getPrimaryComponentAttribute(ACCESSIBILITY_HINT_ATTRIBUTE)).toEqual(AVATAR_ACCESSIBILITY_HINT);
+  it('Set accessibilityHint -> Validate set accessibilityHint is correct', async () => {
+    await expect(
+      await AvatarPageObject.compareAttribute(AvatarPageObject._primaryComponent, Attribute.AccessibilityHint, AVATAR_ACCESSIBILITY_HINT),
+    ).toBeTrue();
+
     await expect(await AvatarPageObject.didAssertPopup()).toBeFalsy(AvatarPageObject.ERRORMESSAGE_ASSERT);
   });
 
-  it('Validate accessibilityRole', async () => {
-    await expect(await AvatarPageObject.getPrimaryComponentAttribute(ACCESSIBILITY_ROLE_ATTRIBUTE)).toEqual(ACCESSIBILITY_ROLE_LINK);
+  it('Set accessibilityRole -> Validate set accessibilityRole is correct', async () => {
+    await expect(
+      await AvatarPageObject.compareAttribute(AvatarPageObject._primaryComponent, Attribute.AccessibilityRole, LINK_A11Y_ROLE),
+    ).toBeTrue();
+
     await expect(await AvatarPageObject.didAssertPopup()).toBeFalsy(AvatarPageObject.ERRORMESSAGE_ASSERT);
   });
 
-  it('Validate default accessibilityRole', async () => {
-    await expect(await AvatarPageObject.getSecondaryComponentAttribute(ACCESSIBILITY_ROLE_ATTRIBUTE)).toEqual(ACCESSIBILITY_ROLE_IMAGE);
+  it('Do NOT set accessibilityRole -> Validate default accessibilityRole is correct', async () => {
```

Comment:
The user does have the ability to set the accessibilityRole. See [useAvatar.ts](https://github.com/microsoft/fluentui-react-native/blob/main/packages/components/Avatar/src/useAvatar.ts#L80):

`accessibilityRole: accessibilityRole ?? 'image'`

I believe most (if not all) of the FURN controls have the accessibilityRole prop exposed (coming from View props) but not all of them allow the default accessibilityRole to be overridden. I'm not sure if that's an oversight or by design.

I think allowing the default accessibilityRole prop to be overridden makes sense for some controls but not all.  A user of Text, for example, should be able to set is as either a 'text', 'link', or 'header' role which are all valid and makes a difference in how it's exposed to screen readers. On the other hand, it may only make sense for a RadioButton to be set as a 'radio' and nothing else. I don't think there should be an issue with letting the user override the accessibilityRole on a control if they want as long as we provide the correct accessibilityRole in the default case so it works properly "out of the box" and with any other role(s) that make sense.

For Avatar, I've seen it be represented as an 'image' or 'button' in different scenarios and I think (according to the spec) there's plans to add an onClick() handler in which case setting it as a 'button' role would make sense. As far as testing goes, I think as long as it makes sense to have a role other than the default role set, we should test our ability to set a different role.

Reply:
Thanks for the info!